### PR TITLE
Prevent redirect_to from logging the token value

### PR DESCRIPTION
**Pull request by omgitsbillryan**
The default rails logging prints the following when `redirect_to` is used (`INFO` level):
```
Redirected to http://www.vets.gov/auth/saml/login?token=abcd1234...
```
This behavior comes from here:
https://github.com/rails/rails/blob/v4.2.7.1/actionpack/lib/action_controller/log_subscriber.rb#L41

This PR unsubscribes this listener and registers my own LogSubscriber that filters this token based on the base url location.

### CODE REVIEW

File: config/initializers/saml_settings.rb [hunk 1]
```diff
@@ -1,2 +0,0 @@
-# frozen_string_literal: true
-SAML_CONFIG = Rails.application.config_for(:saml).freeze
```
Comment (omgitsbillryan) [inline config/initializers/saml_settings.rb]: This moved to config/initializers/override_redirect_to_logging.rb:18 since I needed to use the SAML_CONFIG.

File: spec/controllers/v0/sessions_controller_spec.rb [hunk 2]
```diff
@@ -36,18 +36,19 @@
...
-        expect(Rails.logger).to receive(:error).exactly(1).times
-        expect(post(:saml_callback)).to redirect_to(SAML_CONFIG['relay'] + '?auth=fail')
-        expect(response).to have_http_status(:found)
-      end
       it 'creates a valid session and user' do
         post :saml_callback
         expect(User.find(uuid)).to_not be_nil
         expect(User.find(uuid).attributes).to eq(User.from_merged_attrs(loa1_user, loa3_user).attributes)
       end
+      context ' when SAMLResponse is invalid' do
```
Comment (omgitsbillryan) [inline spec/controllers/v0/sessions_controller_spec.rb]: This is a bit of an unrelated refactor.  I thought the SAML Response being invalid warranted its own context so it may be reused.

**Comment by omgitsbillryan** [comment]
This was a bit tricky to implement and effectively the same could be accomplished with:
```
Rails.logger.silence do
  redirect_to SAML_CONFIG['relay'] + '?token=' + @session.token
end
```

The downside being that any error logging from this call would also get silenced.

**Comment by kreek** [comment]
LGTM 🚀  (and I think this way is better than the `Rails.logger.silence` alternative)
